src/openvic-simulation/map/Map.cpp

```diff
@@ -670,3 +670,82 @@ bool Map::generate_and_load_province_adjacencies(std::vector<LineObject> const&
 	);
 	return ret;
 }
+
+bool Map::load_climate_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary([this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid climate identifier - empty!");
+			return false;
+		}
+
+		Climate* cur_climate = climates.get_item_by_identifier(identifier);
+		if (cur_climate == nullptr) {
+			ModifierValue values;
+			bool ret = modifier_manager.expect_modifier_value(move_variable_callback(values))(node);
+			ret &= climates.add_item({ identifier, std::move(values) });
+			return ret;
+		} else {
+			bool ret = expect_list([this, &identifier, &cur_climate](ast::NodeCPtr list_element) -> bool {
+				std::string_view cur_province_id;
+				bool ret = expect_identifier_or_string(assign_variable_callback(cur_province_id))(list_element);
+				Province* cur_province = provinces.get_item_by_identifier(cur_province_id);
+				if (cur_province != nullptr) {
+					cur_climate->add_province(cur_province);
+					return ret;
+				} else {
+					Logger::error("Invalid province id ", cur_province_id, " in definition of climate ", identifier, "!");
+					return false;
+				}
+			})(node);
+			return ret;
```

Comment:
```suggestion
			return expect_list(expect_province_identifier(
				[cur_climate](Province const& province) {
					cur_climate->add_province(&province));
					return true;
				}
			))(node);
```
`expect_province_identifier` can be put directly into `expect_list`, since it returns a `node_callback_t` and `expect_list` takes it as its argument. `add_province` will deal with any duplicates *in the same climate*, returning false and printing warning messages, but we don't want to return that false value because it will mark `load_climate_file` as failing, when this should only be considered a minor issue that doesn't affect loading the rest of the climate.

This does not address the issue of duplicates across multiple climates, for that I was thinking you could record each province read from a climate list in a single `std::set<Province const*>`, checking if it `contains` each province you find in each climate list and if so not adding the province to that climate, otherwise adding it to the climate and the std::set `std::set<Province const*>`. This would require the lists to be read in reverse order, so that later lists have priority over earlier ones.